src/modals/MarkModal.tsx

```diff
@@ -15,68 +25,11 @@ interface PropsMarkModal {
 const MarkModal = (props: PropsMarkModal) => {
   const { t } = useTranslation();
   const datetimeRef = useRef<null | HTMLIonDatetimeElement>(null);
-  const [disableSave, setDisableSave] = useState(true);
 
   const { cycles, updateCycles } = useContext(CyclesContext);
-  const lengthOfPeriod = useAverageLengthOfPeriod();
-  let markPeriodDays: string[] = [];
 
-  const nowDate = new Date();
-  nowDate.setHours(0, 0, 0, 0);
-
-  const nextCycleFinish: Date = new Date();
-  nextCycleFinish.setDate(nowDate.getDate() + lengthOfPeriod);
-  nextCycleFinish.setHours(0, 0, 0, 0);
-
-  function isLastPeriodDay(date: Date) {
-    date.setHours(0, 0, 0, 0);
-
-    return cycles.some((cycle) => {
-      const startOfCycle = new Date(cycle.startDate);
-      startOfCycle.setHours(0, 0, 0, 0);
-      const endOfCycle = new Date(cycle.startDate);
-      endOfCycle.setHours(0, 0, 0, 0);
-      endOfCycle.setDate(endOfCycle.getDate() + cycle.periodLength);
-      return date >= startOfCycle && date < endOfCycle;
-    });
-  }
-
-  function nextPeriodDays() {
-    const periodDates: string[] = [];
-    if (cycles.length !== 0) {
-      const endOfCurrentCycle = new Date(cycles[0].startDate);
-      endOfCurrentCycle.setDate(
-        endOfCurrentCycle.getDate() + cycles[0].periodLength,
-      );
-      endOfCurrentCycle.setHours(0, 0, 0, 0);
-      if (endOfCurrentCycle >= nowDate) {
-        return undefined;
-      }
-    }
-
-    for (let day = 0; day < (lengthOfPeriod || 5); day++) {
-      const periodDay = new Date(nowDate);
-      periodDay.setHours(0, 0, 0, 0);
-      periodDay.setDate(periodDay.getDate() + day);
-
-      periodDates.push(format(periodDay, "yyyy-MM-dd"));
-    }
-
-    return periodDates;
-  }
-
-  const isActiveDates = (dateString: string) => {
-    if (cycles.length === 0) {
-      return true;
-    }
-    const date = new Date(dateString);
-    date.setHours(0, 0, 0, 0);
-
-    const lastCycleFinish: Date = new Date(cycles[0].startDate);
-    lastCycleFinish.setDate(lastCycleFinish.getDate() + cycles[0].periodLength);
-    lastCycleFinish.setHours(0, 0, 0, 0);
-
-    return date.getTime() > lastCycleFinish.getTime();
+  const isActiveDates = (date: string) => {
```

Comment:
Do we really need this function? You ca just call function from CalculationLogic instead

Reply:
These calculations have been moved to ```CalculationLogic```. This function calls the ```getMarkModalActiveDates``` function from ```CalculationLogic```.
But after the comment, I removed this function. Now the ```getMarkModalActiveDates``` function is calculated directly from
```
isDateEnabled={(date: string) => {
               return getMarkModalActiveDates(date, cycles);
             }}
```